server/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -299,6 +299,10 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
         if (sec != null) {
             return doSearchRewrite(sec);
         }
+        final QueryRewriteContext context = queryRewriteContext.convertToMappingMetadataAwareContext();
```

Comment:
I am wondering if this if statement should go after `convertToCoordinatorRewriteContext` but before `convertToSearchExecutionContext`. Rewriting without an index searcher happens before we have a `SearhcExecutionContext`...

Reply:
I picked this order because it was suggested this way in [this comment](https://github.com/elastic/elasticsearch/pull/96161#issuecomment-1557287181).

Also the real order depends on which what kind of context the doRewrite() method is called?